research/2d-decoding/rectangle_detector.py

```diff
@@ -103,42 +114,47 @@ def readImage(filename):
         nottaSquare = False
         # Find the pixels with the biggest and smallest x and y values, if more than 1 pixel have the smallest x, then take the one with the smallest y
         # Biggest y we take the smallest x, biggest x we take the biggest y and smallest y we take the biggest x (basically rotating clockwise)
-        minX, maxX = min(mins(group, key = lambda a: a[0]), key = lambda a: a[1]), max(maxs(group, key = lambda a: a[0]), key=lambda a: a[1])
-        minY, maxY = max(mins(group, key = lambda a: a[1]), key=lambda a: a[0]), min(maxs(group, key = lambda a: a[1]), key = lambda a: a[0])
+        minX, maxX = min(mins(group, key=lambda a: a[0]), key=lambda a: a[1]), max(
+            maxs(group, key=lambda a: a[0]), key=lambda a: a[1])
+        minY, maxY = max(mins(group, key=lambda a: a[1]), key=lambda a: a[0]), min(
+            maxs(group, key=lambda a: a[1]), key=lambda a: a[0])
 
         # Check that all the points are different
         if not minX != maxX != minY != maxY:
             continue
 
         # Line equations are y = m(x-a)+b, with A(a, b) being on the line and m being the slope
         try:
-            line = lambda a: (maxY[1]-maxX[1])/(maxY[0]-maxX[0])*(a - maxX[0]) + maxX[1] # Lower right side
+            # Lower right side
+            def line(a): return (maxY[1]-maxX[1]) / \
+                (maxY[0]-maxX[0])*(a - maxX[0]) + maxX[1]
             for x in range(maxY[0], maxX[0]+1):
                 y = round(line(x))
                 for offset in range(-LINE_PIXEL_TOLERANCE, LINE_PIXEL_TOLERANCE+1):
                     if y+offset < 0 or y+offset >= len(pixels):
                         continue
                     if pixels[y+offset][x] == 0:
                         break
-                else: # If no pixels within the tolerance are white, then the line isn't continuous, so this group isn't a rectangle
+                else:  # If no pixels within the tolerance are white, then the line isn't continuous, so this group isn't a rectangle
                     nottaSquare = True
                     break
         except ZeroDivisionError:
             continue
         if nottaSquare:
             continue
-        
+
         try:
             if minX[0]-maxY[0] != 0:
-                line = lambda a: (minX[1]-maxY[1])/(minX[0]-maxY[0])*(a - minX[0]) + minX[1] # Lower left side
+                def line(a): return (
+                    minX[1]-maxY[1])/(minX[0]-maxY[0])*(a - minX[0]) + minX[1]  # Lower left side
```

Comment:
readability ?

lack of consistency with the other func, also why is the paren on one line, and everything else on another?

Reply:
Outdated + couldn't find the code in my codebase